src/jit/lowerxarch.cpp

```diff
@@ -880,6 +902,114 @@ void Lowering::LowerSIMD(GenTreeSIMD* simdNode)
 //
 void Lowering::LowerHWIntrinsic(GenTreeHWIntrinsic* node)
 {
+    switch (node->gtHWIntrinsicId)
+    {
+        case NI_SSE_CompareEqualOrderedScalar:
+            LowerHWIntrinsicCC(node, NI_SSE_COMISS, GenCondition::FEQ);
+            break;
+        case NI_SSE_CompareNotEqualOrderedScalar:
+            LowerHWIntrinsicCC(node, NI_SSE_COMISS, GenCondition::FNEU);
+            break;
+        case NI_SSE_CompareLessThanOrderedScalar:
+            LowerHWIntrinsicCC(node, NI_SSE_COMISS, GenCondition::FLT);
+            break;
+        case NI_SSE_CompareLessThanOrEqualOrderedScalar:
+            LowerHWIntrinsicCC(node, NI_SSE_COMISS, GenCondition::FLE);
+            break;
+        case NI_SSE_CompareGreaterThanOrderedScalar:
+            LowerHWIntrinsicCC(node, NI_SSE_COMISS, GenCondition::FGT);
+            break;
+        case NI_SSE_CompareGreaterThanOrEqualOrderedScalar:
+            LowerHWIntrinsicCC(node, NI_SSE_COMISS, GenCondition::FGE);
+            break;
+
+        case NI_SSE_CompareEqualUnorderedScalar:
```

Comment:
"Design by committee" intrinsic names. They give the impression that the condition is "unordered" but in fact it's the compare instruction itself that it "unordered" (aka quiet, aka non-signaling).

Reply:
Are you suggesting these be named `UnorderedCompareEqualScalar` instead? Or possibly some other name?

Reply:
Yep, `UnorderedCompareEqualScalar` is a reasonable option because it's closer to the native intrinsic name and instruction name. Another option would be to not include the ordered ones because there's no obvious case for them in .NET.

Reply:
> because there's no obvious case for them in .NET.

Right, and I believe last time I discussed this with @CarolEidt, we determined that there wasn't any pressing need to support floating-point exceptions (even as opt-in) for .NET. For `COMISS` and `UCOMISS` basically the only difference is that the former also throws for `QNaN` (when floating-point exceptions are enabled).

>  Another option would be to not include the ordered ones

I think a downside here is that we can't just call it `CompareEqualScalar` since you can't overload by return type (and `CMPSS` currently holds that name). Prefixing it with `Unordered` (and possibly `Ordered` for the COMISS` ones) also makes them less discoverable. Maybe `CompareOrderedEqualScalar` and `CompareUnorderedEqualScalar` would help keep them discoverable....

I'll split this out into a separate issue for further discussion.